pkg/rc/replication_controller.go

```diff
@@ -236,6 +242,12 @@ func (rc *replicationController) WatchDesires(quit <-chan struct{}) <-chan error
 		}
 
 		for rcFields := range rcChanges {
+			select {
+			case <-shouldInitialCheck:
+				rc.checkMissingArtifacts(rcFields)
+			case <-shouldPeriodicCheck:
+				rc.checkMissingArtifacts(rcFields)
```

Comment:
as I said in my last comment i'd probably do a `shouldPeriodicCheck.Reset(time.Hour)` here